bucketcopy/activity_test.go

```diff
@@ -0,0 +1,85 @@
+package bucketcopy_test
+
+import (
+	"context"
+	"testing"
+
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_testsuite "go.temporal.io/sdk/testsuite"
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/memblob"
+	"gotest.tools/v3/assert"
+
+	"github.com/artefactual-sdps/temporal-activities/bucketcopy"
+)
+
+func bucket(t *testing.T, key, contents string) *blob.Bucket {
+	t.Helper()
+
+	b := memblob.OpenBucket(nil)
+	t.Cleanup(func() { b.Close() })
+
+	if key != "" && contents != "" {
+		b.WriteAll(context.Background(), key, []byte(contents), nil)
+	}
+
+	return b
+}
+
+func TestActivity(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		bucket  *blob.Bucket
+		params  bucketcopy.Params
+		wantErr string
+	}{
+		{
+			name:   "Copies a blob",
+			bucket: bucket(t, "source.txt", "content"),
+			params: bucketcopy.Params{
+				SourceKey: "source.txt",
+				DestKey:   "dest.txt",
+			},
+		},
+		{
+			name:   "Fails copying a missing blob",
+			bucket: bucket(t, "", ""),
+			params: bucketcopy.Params{
+				SourceKey: "missing.txt",
+				DestKey:   "dest.txt",
+			},
+			wantErr: "bucketcopy: copy blob:",
```

Comment:
Does the error contain the name of the source file? I think that would be helpful for debugging.

Reply:
Yes, the actual error is something like:

```
bucketcopy: copy blob: blob (key "missing.txt -> dest.txt") (code=NotFound): blob not found
```

I'm just checking the parts the activity adds to that error.